incubator/gocd/templates/gocd-server-deployment.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "gocd.fullname" . }}-server
+  labels:
+    app: {{ template "gocd.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+    component: {{ template "gocd.name" . }}-server
+spec:
+  {{- if le .Values.server.replicaCount 1.0 }}
+  replicas: {{ .Values.server.replicaCount }}
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "gocd.name" . }}
+        release: {{ .Release.Name | quote }}
+        component: {{ template "gocd.name" . }}-server
+    spec:
+      {{- if .Values.server.persistence.enabled }}
+      volumes:
+        - name: goserver-vol
+          persistentVolumeClaim:
+            claimName: {{ template "gocd.fullname" . }}
+      {{- end }}
+      containers:
+        - name: {{ template "gocd.name" . }}-server
+          {{- if .Values.server.image.tag }}
+          image: "{{ .Values.server.image.repository }}:{{ .Values.server.image.tag }}"
+          {{- else }}
+          image: "{{ .Values.server.image.repository }}:v{{ default .Chart.AppVersion .Values.app.version }}"
+          {{- end }}
+          imagePullPolicy: {{ .Values.server.image.pullPolicy }}
+          env:
+            {{- if .Values.server.env.goServerSystemProperties }}
+            - name: GO_SERVER_SYSTEM_PROPERTIES
+              value: {{ .Values.server.env.goServerSystemProperties }}
+            {{- end }}
+{{ toYaml .Values.server.env.extraEnvVars | indent 12 }}
```

Comment:
@GaneshSPatil I tried running this chart on our cluster with a `goServerSystemProperties` value, and I faced an issue with this line. A`null` is spit out causing the generated YAML to be incorrect. To fix, I modified this line to:

```
{{- if .Values.server.env.extraEnvVars }}
  {{ toYaml .Values.server.env.extraEnvVars | indent 12 }}
{{- end }}
```

A snippet from the configuration I used:

```
server:
  # other configurations
  # ...
  env:
    goServerSystemProperties: "-Xmx1024m"
    extraEnvVars:
```